### PR TITLE
improve grant handling for space members

### PR DESCRIPTION
**Pull request by micbar**
# Description

Improve the grant handling

### CODE REVIEW

File: internal/grpc/services/gateway/usershareprovider.go [hunk 1]
```diff
@@ -698,10 +698,19 @@ func (s *svc) removeShare(ctx context.Context, req *collaboration.RemoveShareReq
 
 func (s *svc) removeSpaceShare(ctx context.Context, ref *provider.ResourceId, grantee *provider.Grantee) (*collaboration.RemoveShareResponse, error) {
 	listGrantRes, err := s.listGrants(ctx, ref)
+	var permissions *provider.ResourcePermissions
```
Comment (kobergj) [inline internal/grpc/services/gateway/usershareprovider.go:701]: can you move this line down? It doesn't make sense to have `permissions` created here before the `err` from `listGrants` is checked...

File: internal/grpc/services/gateway/usershareprovider.go [hunk 2]
```diff
@@ -733,3 +742,23 @@ func shareIsSpaceRoot(key *collaboration.ShareKey) bool {
 	}
 	return refIsSpaceRoot(key.ResourceId)
 }
+
+func isEqualGrantee(a, b provider.Grantee) bool {
```
Comment (kobergj) [inline internal/grpc/services/gateway/usershareprovider.go:746]: Would it make sense to have a and b being `*provider.Grantee`? We would avoid having to dereference them in line `706`.

**Review by kobergj — CHANGES_REQUESTED** [review]
LG :+1: Just some minors

File: internal/grpc/services/gateway/usershareprovider.go (hunk 1 above)
Comment (micbar) [reply internal/grpc/services/gateway/usershareprovider.go]: done

File: internal/grpc/services/gateway/usershareprovider.go (hunk 2 above)
Comment (micbar) [reply internal/grpc/services/gateway/usershareprovider.go]: done

**Review by kobergj — APPROVED** [review]

**Pull request merged**